wolfssl/wolfcrypt/port/Espressif/esp32-crypt.h

```diff
@@ -22,13 +22,143 @@
 
 #define __ESP32_CRYPT_H__
 
-#include "wolfssl/wolfcrypt/settings.h"
-#include <wolfssl/wolfcrypt/types.h> /* for MATH_INT_T */
-
-#include "esp_idf_version.h"
-#include "esp_types.h"
-#include "esp_log.h"
-
+#include "sdkconfig.h" /* ensure ESP-IDF settings are available everywhere */
+
+/* wolfSSL  */
+#include <wolfssl/wolfcrypt/settings.h> /* references user_settings.h */
+#include <wolfssl/wolfcrypt/types.h>    /* for MATH_INT_T */
+
+/* Espressif */
+#include <esp_idf_version.h>
+#include <esp_types.h>
+#include <esp_log.h>
+
+/* exit codes to be used in tfm.c, sp_int.c, integer.c, etc. */
+/* TODO what numbers do we really want to use? */
+#define MP_HW_ERROR    (-106) /* hardware error, consider SW fallback  */
```

Comment:
can WC_HW_WAIT_E and WC_HW_E be used?

Reply:
oh, good suggestion. The ` WC_HW_E` is fairly generic, whereas the `MP_HW_ERROR` is specific to the math acceleration.

I'm ok with either. Which do you prefer?

This is related to the above code review comment and https://github.com/wolfSSL/wolfssl/pull/6565

Reply:
I think we should be using the error-crypt.h errors over making a new section for errors that need to be maintained and searched out when comparing return values. Additional errors could be appended to that list if needed. For more verbose error messages though please use the WOLFSSL_MSG() debug macros. For example we have a lot of sections of code that can trigger a BAD_FUNC_ARG error or MEMORY_E, which is pretty generic, and then the issue can be debugged farther in debug mode with additional WOLFSSL_MSG() logs.

Reply:
The ESP specific fallback error messages can stay here in esp32-crypt.h. Thanks for looking for which error messages could be reused from error-crypt.h.

Reply:
I've taken your suggestion for `WC_HW_E` instead of `MP_HW_ERROR` and `WC_HW_WAIT_E` instead of `MP_HW_BUSY`.

I'll keep `MP_HW_FALLBACK` that is more of a state hint than an error (e.g. "_all is ok, but you need to fall back to SW_").

I'll also keep `MP_HW_VALIDATION_ACTIVE` that is only useful during debugging. This too is an indication that a SW calc is explicitly needed to compare to a prior HW calc for comparison.

Both will remain in `error-crypt.h` with addition comments regarding usage.